### PR TITLE
Plugin Facelift

### PR DESCRIPTION
**Pull request by oleg-nenashev**
[x] Jenkinsfile
[x] Update Parent POM and core dependency
[x] Cleanup issues in the code
[x] Realize how bad my English was 4 years ago, fix some bits

@jenkinsci/code-reviewers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
 
 License
 --------
-[MIT License][1]
-
-[1]: http://www.opensource.org/licenses/mit-license.php
-[2]: https://github.com/oleg-nenashev
-[3]: http://www.jenkinsci.org/
-[4]: https://wiki.jenkins-ci.org/display/JENKINS/Extra+Tool+Installers+Plugin
+[MIT License][http://www.opensource.org/licenses/mit-license.php]
```
Comment (jglick) [inline README.md]: prefer: https://opensource.org/licenses/MIT

File: pom.xml [hunk 2]
```diff
@@ -15,34 +15,31 @@
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Extra+Tool+Installers+Plugin</url>
 
     <properties>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <jenkins.version>1.651.3</jenkins.version>
```
Comment (jglick) [inline pom.xml]: Going to jump all the way to 1.651.x but not just go to Jenkins 2?

File: pom.xml [hunk 3]
```diff
@@ -15,34 +15,31 @@
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Extra+Tool+Installers+Plugin</url>
```
Comment (jglick) [inline pom.xml]: BTW I like to rewrite these to the current canonical URL.

File: src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/utils/EnvStringParseHelper.java [hunk 4]
```diff
@@ -34,7 +39,7 @@
      * Resolves tools installation directory using global variables.
      * @param environment Collection of environment variables
      * @param macroString Input path with macro calls
-     * @return Raw string
+     * @return  Raw string
```
Comment (jglick) [inline src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/utils/EnvStringParseHelper.java]: ?

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [reply pom.xml]: Yes for now. I am going to jump to 2.x once I am ready to update to parent POM 3.0 x

File: pom.xml (hunk 3 above)
Comment (oleg-nenashev) [reply pom.xml]: Will do it before the release. @daniel-beck Is it already fine to reference plugins.jenkins.io instead of Wiki?

Comment (jglick) [reply pom.xml]: I was rather referring to: https://wiki.jenkins.io/display/JENKINS/Extra+Tool+Installers+Plugin

**Pull request merged**